sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -793,6 +794,20 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
       orderedPartitionSpec.put(colName.toLowerCase, partition(colName))
     }
 
+    // Access SQLConf to get 'Dynamic Partition Inserts' parameter specified dynamically
+    // after HiveClient is created
+    val sqlConf = SparkSession.getActiveSession.get.sessionState.conf
+    Seq(
+      "hive.exec.max.dynamic.partitions",
+      "hive.exec.max.dynamic.partitions.pernode",
+      "hive.exec.max.created.files",
+      "hive.error.on.empty.partition"
+    ).foreach { param =>
+      if (sqlConf.contains(param)) {
+        client.runSqlHive(s"set $param=${sqlConf.getConfString(param)}")
```

Comment:
That will be the best approach since this is a general issue for all unhandled hive param options. The reason to do this here is that `SetCommand` lives in `sql/core` and does not interact with this. Is there a way to invoke `runSqlHive` there?